File: framework/languages/dotnet/e2e/YieldDispatch/Server/Delay/DelayHostFactory.cs
using Systems.Zlink;
using YieldDispatch.Shared;
using Zlink.Framework.AspNetCore;

namespace YieldDispatch.Server.Delay;

internal static class DelayHostFactory
{
    public static WebApplication Create(string[] args)
    {
        var options = DelayOptions.Parse(args);
        Directory.CreateDirectory(options.LogDir);

        var builder = WebApplication.CreateBuilder(args);
        builder.Logging.ClearProviders();
        builder.Logging.AddSimpleConsole(console =>
        {
            console.SingleLine = true;
            console.TimestampFormat = "HH:mm:ss.fff ";
        });
        builder.WebHost.UseUrls(options.HttpUrl);
        builder.Services.AddSingleton(new EvidenceStore(options.Rid, options.EvidenceFile));
        builder.Services.AddSingleton(new NodeOptions(options.Rid));
        builder.Services.AddZLinkFramework(framework =>
        {
            framework.AddHandlersFromAssemblyOf(typeof(Program));
            framework.AddClientServerChannel(YieldDispatchNames.DelayChannel)
                .EnableServer(options.DelayEndpoint)
                .SetRoutingId(RoutingId.From(options.Rid))
                .AddRequestHandler<DelayHandler, DelayReq, DelayReply>("DelayReq");
        });

        var app = builder.Build();
        app.MapGet("/health", () => Results.Ok(new { status = "ready", role = "delay", options.Rid }));
        app.MapGet("/evidence", (EvidenceStore evidence) => Results.Ok(evidence.Snapshot()));
        return app;
    }
}

File: framework/languages/node/e2e/YieldDispatch/Server/Delay/delay-host-factory.ts
import { Module } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import { ZLinkModule, zlinkFramework } from '@zlink-systems/nestjs';
import { YieldDispatchNames } from '../../Shared/messages';
import { EvidenceStore } from './Support/evidence-store';
import { closeHttpServer, startHttpServer } from './Support/http-server';
import { parseDelayOptions } from './Configuration/delay-options';
import { DelayHandler } from './Handlers/delay-handler';

export async function startDelayHost(args: readonly string[]): Promise<void> {
  const options = parseDelayOptions(args);
  const evidence = new EvidenceStore(options.rid, options.evidenceFile);
  let stopping = false;

  class DelayModule {}
  Module({
    imports: [
      ZLinkModule.forRootFactory({
        useFactory: () => zlinkFramework()
          .addClientServerChannel(YieldDispatchNames.delayChannel)
            .enableServer(options.delayEndpoint)
            .routingId(options.rid)
            .addRequestHandler('DelayReq', DelayHandler)
          .build()
      })
    ],
    providers: [
      { provide: EvidenceStore, useValue: evidence },
      DelayHandler
    ]
  })(DelayModule);

  const app = await NestFactory.createApplicationContext(DelayModule, { logger: false });
  const server = await startHttpServer(options.httpUrl, [
    { method: 'GET', path: '/health', handle: () => ({ status: 'ready', role: 'delay', rid: options.rid }) },
    { method: 'GET', path: '/evidence', handle: () => evidence.snapshot() },
    {
      method: 'POST',
      path: '/shutdown',
      handle: () => {
        stopping = true;
        return { status: 'stopping' };
      }
    }
  ]);
  while (!stopping) {
    await new Promise((resolve) => setTimeout(resolve, 100));
  }
  await closeHttpServer(server);
  await app.close();
}
